client/src/app/shared/loading-bar/loading-bar.service.ts

```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+
+@Injectable()
+export class LoadingBarService {
+  private event: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
```

Comment:
@Jefiozie can you elaborate why we use a BehaviourSubject here?

Reply:
I thought it would be best to have the visibility on initial always false. But this is not needed here as the component it self is default false. So I changed it to a subject.